### PR TITLE
Add command-line parameter to set the location of the registry file

### PR DESCRIPTION
**Pull request by threadwaste**
The registry file is always written to logstash-forwarder's working directory as a hidden file, .logstash-forwarder. This change makes this file relocatable. The default argument maintains the current behavior.

### CODE REVIEW

**Comment by jordansissel** [comment]
@threadwaste can you do step 2 of the contributing guide? https://github.com/elasticsearch/logstash-forwarder/blob/master/CONTRIBUTING.md#contribution-steps

**Comment by threadwaste** [comment]
@jordansissel Oops. I thought I had sorted that out. Done.

**Comment by jordansissel** [comment]
Hmm; merging fails. Can you rebase? I can do this Monday if you haven't.

**Comment by threadwaste** [comment]
@jordansissel Rebased and pushed. This should merge cleanly, now.

**Comment by zavyrylin** [comment]
+1 Default bahaviour of logstash-forwarder has turned out to be very frustrating for environments where access rights on the workdir are strict.

**Comment by jordansissel** [comment]
Can you add tests for this?

**Comment by threadwaste** [comment]
Happy to add tests, though it seems neither source file modified here has any. Am I overlooking something? If not, I can light the fire for testing both where appropriate.

**Comment by jordansissel** [comment]
Indeed, there are likely no tests in this area right now.

File: logstash-forwarder.go [hunk 1]
```diff
@@ -155,13 +160,15 @@ func main() {
...
 	restart.files = make(map[string]*FileState)
-	if existing, e := os.Open(".logstash-forwarder"); e == nil {
+	if existing, e := os.Open(options.registryFile); e == nil {
 		defer existing.Close()
 		wd := ""
-		if wd, e = os.Getwd(); e != nil {
-			emit("WARNING: os.Getwd retuned unexpected error %s -- ignoring\n", e.Error())
+		if !filepath.IsAbs(options.registryFile) {
+			if wd, e = os.Getwd(); e != nil {
+				emit("WARNING: os.Getwd retuned unexpected error %s -- ignoring\n", e.Error())
```
Comment (msimerson) [inline logstash-forwarder.go]: returned

**Comment by threadwaste** [comment]
@jordansissel Finally had some time to take a first crack at tests. Looking at other PRs and issues, there seems to be active discussion around the CLI parameters and the configuration file. So, I opted to leave the registrar's direct use of the options struct in place. Let me know if I missed the mark.

**Comment by threadwaste** [comment]
Actually, I went ahead and refactored Registrar.

**Comment by ruflin** [comment]
Implemented in filebeat: https://github.com/elastic/filebeat/pull/10

**Comment by jordansissel** [comment]
Thanks for helping make logstash-forwarder better! Looks like elastic/filebeat#10 covers this.

Logstash-forwarder is going away and is replaced by https://github.com/elastic/filebeat/. If this is still an issue with filebeat, would you mind opening a ticket [there](https://github.com/elastic/filebeat/issues)?

**Pull request closed without merging**